portfolio/src/main/java/com/google/sps/servlets/CovidDataServlet.java

```diff
@@ -38,13 +57,40 @@ public void init() {
       String line = scanner.nextLine();
       String[] cells = line.split(",");
 
-      ArrayList<Integer> totals = new ArrayList<>();
+      Date date = new Date();
+      try {
+        date = new SimpleDateFormat("yyyy-MM-dd").parse(cells[0]);
+      } catch (ParseException pex) {
+        pex.printStackTrace();
+        LOGGER.log(Level.WARNING, "invalid date string");
+      }    
 
-      String date = cells[0];
-      totals.add(Integer.valueOf(cells[1]));
-      totals.add(Integer.valueOf(cells[2]));
+      // Validation checks for parsing COVID-19 numbers.
+      int totalCases = 0;
+      int newCases = 0;
+      try {
+        totalCases = Integer.valueOf(cells[1]);
+      } catch (NumberFormatException nfex) {
+        nfex.printStackTrace();
```

Comment:
for these specific exceptions, i dont think we need to printStackTrace given that it's a data reading error might be a bit excessive to print the whole stack trace, imo.